cpp/src/arrow/flight/ArrowFlightConfig.cmake.in

```diff
@@ -34,3 +34,5 @@ include("${CMAKE_CURRENT_LIST_DIR}/ArrowFlightTargets.cmake")
 arrow_keep_backward_compatibility(ArrowFlight arrow_flight)
 
 check_required_components(ArrowFlight)
+
+arrow_show_details(arrow_flight)
```

Comment:
If a project imports multiple components from arrow, like `ArrowFlight`, `ArrowCUDA` and so on, these messages tend to get repetitive as these all inherently the require Arrow core, and thus `arrow_show_details(arrow)` is called multiple times.

Reply:
I think that it's not a problem but we can suppress it by defining a variable that keeps whether we already showed these messages or not:

```cmake
function(arrow_show_details package_name)
  if(${package_name}_SHOWED_DETAILS)
    return()
  endif()
  # ...
  set(${package_name}_SHOWED_DETAILS TRUE)
endfunction()
```

(We may need to a macro instead of a function to use the variable over function calls. If we use a macro, we can't use `return()` here: https://cmake.org/cmake/help/latest/command/return.html